openalex-importer/src/db/index.ts

```
@@ -51,291 +81,335 @@ const {
 
-export const createBatch = async (tx: PgTransactionType, queryInfo: QueryInfo) => {
-  const savedBatch = await tx.insert(batchesInOpenAlex).values(queryInfo).returning({ id: batchesInOpenAlex.id });
-  return savedBatch[0].id;
-};
+export const finalizeBatch = async (tx: pgPromise.ITask<any>, batchId: number) =>
+  await tx.none('UPDATE openalex.batch SET finished_at = $1 WHERE id = $2', [new UTCDate(), batchId]);
 
-export const finalizeBatch = async (tx: PgTransactionType, batchId: number) =>
-  await tx.update(batchesInOpenAlex).set({ finished_at: new UTCDate() }).where(eq(batchesInOpenAlex.id, batchId));
+export const saveData = async (tx: pgPromise.ITask<any>, batchId: number, models: DataModels) => {
+  const counts = Object.entries(models).reduce((acc, [k, a]) => ({ ...acc, [k]: a.length}), {});
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)